src/ArduinoLogger.h

```diff
@@ -270,6 +279,26 @@ class LoggerBase
 #endif
 	}
 
+#ifdef __XTENSA__
+	/// Prints directly to the log with no extra characters added to the message.
+	void print(const char* format, ...) noexcept
+	{
+		// We need to capture this with a lambda to call log_putc,
+		// Then decay that into a function pointer for a callback
+		auto f = [this](char c) { log_putc(c); };
+		void (*callback)(char) = Lambda::ptr(f);
+
+		va_list va;
+		va_start(va, format);
+		ets_vprintf(callback, format, va);
```

Comment:
This line would cause compilation error if in ets_sys.h function declaration is as follows:
`int ets_vprintf(int (*print_function)(int), const char * format, va_list arg) __attribute__ ((format (printf, 2, 0)));`
and this is the case for ArduCAM ESP8266 UNO
error:
`arduino-logger/src/ArduinoLogger.h:293:35: error: invalid conversion from 'void (*)(char)' to 'int (*)(int)' [-fpermissive]
   ets_vprintf(callback, format, va);
                                   ^`